### PR TITLE
Add clear cache api

### PR DESCRIPTION
**Pull request by amanjain97**
This PR is regarding adding `_cache/clear` API to elastic client. 
Doc: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/indices-clearcache.html

Please suggest if anything else is required.

### CODE REVIEW

File: indices_clear_cache.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
+package elastic
```
Comment (olivere) [inline indices_clear_cache.go:5]: The copyright notice is missing here.

**Review by olivere — CHANGES_REQUESTED** [review]
The copyright notice is missing in the file headers, but the code looks good.

Please double-check that the service has all properties defined in the [REST API spec](https://github.com/elastic/elasticsearch/blob/6.7/rest-api-spec/src/main/resources/rest-api-spec/api/indices.clear_cache.json) for v6. It looks complete, but I haven't checked in depth.

Thanks!

**Comment by olivere** [comment]
I will also need to port it to v7 as well.

File: indices_clear_cache.go (hunk 1 above)
Comment (amanjain97) [reply indices_clear_cache.go:5]: @olivere Sorry, I am not much familiar with adding copyrights. I have added my name. Please check.

**Comment by amanjain97** [comment]
Can you take a look at https://github.com/elastic/elasticsearch/blob/6.7/rest-api-spec/src/main/resources/rest-api-spec/api/indices.clear_cache.json#L49

I added `request_cache` field which seems duplicated with `request`.

File: indices_clear_cache.go (hunk 1 above)
Comment (olivere) [reply indices_clear_cache.go:5]: No. Simply copy the header from any of the other files.

Comment (amanjain97) [reply indices_clear_cache.go:5]: Sure ok.

**Comment by olivere** [comment]
> Can you take a look at https://github.com/elastic/elasticsearch/blob/6.7/rest-api-spec/src/main/resources/rest-api-spec/api/indices.clear_cache.json#L49
> 
> I added `request_cache` field which seems duplicated with `request`.

Oh, one of them seems to be a deprecated field. I typically look up in the Java code which of the fields is used. The high-level Java client [seems to use the `request` field](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestClearIndicesCacheAction.java#L65).

**Comment by amanjain97** [comment]
@olivere I shall add comment on the function as deprecated. Maybe we can remove in v7

**Comment by olivere** [comment]
Will be part of the next v6 and v7 release. Thank you.

**Comment by amanjain97** [comment]
Sorry, why did you close when you write `Will be part of the next v6 and v7 release. Thank you.`

**Comment by olivere** [comment]
Because it's already part of the release branches, but not tagged yet.

**Pull request closed without merging**